modules/setting-up-an-openshift-login.adoc

```diff
@@ -0,0 +1,90 @@
+// Module is included in the following assemblies:
+//
+// * openshift-docs/cicd/gitops/getting-started--with-openshift-gitops.adoc
+
+[id="setting-up-an-openshift-login_{context}"]
+= Setting up an OpenShift login
+
+== Creating a new client in Keycloak
+
+To set up an OpenShift login, you must create a new client in Keycloak:
+
+. Log in to your Keycloak server.
+. Navigate to *myrealm* -> *Clients* ->  and then select *Create*:
+.. For *Client ID*, type *argocd*.
+.. For *Client Protocol*, select *openid-connect*.
+.. For *Root URL*, select the Argo CD route URL.
+. Configure the client:
+.. For *Access Type*, select *confidential*.
+.. For *Valid Redirect URLs*, select the callback url for your Argo CD hostname. Your callback should be `https://{hostname}/auth/callback`.
+.. Set the *Base URL* to */applications*.
+.. Save your work and then you get the *Credentials* tab. Go to this tab.
+.. For *Client Authenticator*, select *Client ID and Secret* and copy the *argocd-secret* so you can use it for your Argo CD configuration.
+
+== Configuring Argo CD OpenID Connect (OIDC)
+
+After you create a new client in Keycloak and generate an *argocd-secret* in the Argo CD secret, you get a client secret. To configure Argo CD OIDC, store this client secret with the following instructions:
+
+. Encode the client secret in base64:
++
+[source,terminal]
+----
+$ echo -n '83083958-8ec6-47b0-a411-a8c55381fbd2' | base64
+----
+
+. Add the base64 value to a new key called `oidc.keycloak.clientSecret` and edit the secret:
++
+[source,terminal]
+----
+$ kubectl edit secret argocd-secret -n argocd
+----
+Your secret now looks like the following:
++
+[source,terminal]
+----
+$ yaml apiVersion: v1
+kind: Secret
+metadata: name: argocd-secret
+data:
+  oidc.keycloak.clientSecret: ODMwODM5NTgtOGVjNi00N2IwLWE0MTEtYThjNTUzODFmYmQy
+----
+. To enable your Keycloak authentication, add the OIDC configuration and edit the Argo CD CR:
++
+[source,terminal]
+----
+$ kubectl edit argocd -n argocd
+----
+Your Argo CD CR now looks like the following:
++
+[source,terminal]
+----
+  $ apiVersion: argoproj.io/v1alpha1
+kind: ArgoCD
+metadata:
+  creationTimestamp: null
+  name: argocd
+  namespace: argocd
+spec:
+  resourceExclusions: |
+    - apiGroups:
+      - tekton.dev
+      clusters:
+      - '*'
+      kinds:
+      - TaskRun
+      - PipelineRun
+  oidcConfig: |
+    name: OpenShift Single Sign-On
+    issuer: https://keycloak.example.com/auth/realms/myrelam
+    clientID: argocd
+    clientSecret: $oidc.keycloak.clientSecret
+    requestedScopes: ["openid", "profile", "email", "groups"]
+  server:
+    route:
+      enabled: true
+----
+. Verify the following:
```

Comment:
I would not make this a separate step. I'd make these checks annotations on the YAML itself. This format is hard to follow.